From 1e3c4163cab4b31d19a18b65534da4606267f13b Mon Sep 17 00:00:00 2001
From: Praveen Diwakar <praveen.diwakar@intel.com>
Date: Tue, 21 Jul 2015 05:11:23 +0530
Subject: [PATCH] ASoC: Intel: Set FBA probe gain to default values

(Frame Based Algorithmn) FBA probes are implemented using 6 gains.
Gain[0]: ProbeIn -> SmBufIdIn
Gain[1]: SmBufIdOut -> ProbeIn|SmBufIdOut Mixing point
Gain[2]: ProbeIn -> ProbeIn|SmBufIdOut Mixing point
Gain[3]: ProbeIn|SmBufIdOut Mixing point -> BufferProbeIn
Gain[4]: ProbeIn|SmBufIdOut Mixing point -> ProbeOut
Gain[5]: BufferProbeOut -> ProbeOut

Change-Id: I9e89154e02eb21e028a4428b8d24c193e5a1782f
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-12548
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 49faf642e8ae..b5eb1535f31e 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -1603,6 +1603,21 @@ static int sst_tone_generator_event(struct snd_soc_dapm_widget *w,
 	return 0;
 }
 
+#define GAIN_MUTE	0xFC40
+#define GAIN_UNMUTE	0x0000
+static void sst_fill_probe_gain(struct sst_cmd_probe *cmd, int mode)
+{
+	if (mode == SST_PROBE_INJECTOR) {
+		cmd->gain[0] = GAIN_MUTE;
+		cmd->gain[1] = GAIN_UNMUTE;
+		cmd->gain[2] = GAIN_MUTE;
+	} else {
+		cmd->gain[0] = GAIN_UNMUTE;
+		cmd->gain[1] = GAIN_MUTE;
+		cmd->gain[2] = GAIN_UNMUTE;
+	}
+}
+
 static int sst_send_probe_cmd(struct sst_data *sst, u16 probe_pipe_id,
 			      int mode, int switch_state,
 			      const struct sst_probe_config *probe_cfg)
@@ -1631,6 +1646,9 @@ static int sst_send_probe_cmd(struct sst_data *sst, u16 probe_pipe_id,
 	cmd.cfg.format = probe_cfg->cfg.format;
 	cmd.sm_buf_id = 1;
 
+	if (switch_state == SST_SWITCH_ON)
+		sst_fill_probe_gain(&cmd, mode);
+
 	return sst_fill_and_send_cmd(sst, SST_IPC_IA_CMD, SST_FLAG_BLOCKED,
 				     probe_cfg->task_id, 0, &cmd,
 				     sizeof(cmd.header) + cmd.header.length);
-- 
1.9.1

